src/Core/src/Platform/Android/MauiAppCompatActivity.Lifecycle.cs

```diff
@@ -18,6 +19,9 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
 			MauiApplication.Current?.Services?.InvokeLifecycleEvents<AndroidLifecycle.OnActivityResult>(del => del(this, requestCode, resultCode, data));
 		}
 
+		// TODO: Investigate whether the new AndroidX way is actually useful:
+		//       https://developer.android.com/reference/android/app/Activity#onBackPressed()
+		[Obsolete]
```

Comment:
@PureWeen this is interesting. The "new" way is to add callbacks to some "dispatcher", and if any callback is marked "enabled" then it does not finish the activity. However, that "enabled" is a property that is set. We use this as a JIT callback to run and it determines whether it wants to end or not.

This breaks our use case where we want to run some code that does work and then decides "nah, I'm gonna stick around". The new way is you need to update the "enabled" state depending on whether or not you want to end - and then let the app call your callback or finish the activity. This is a bit weird because it appears that if you change your mind in the callback, you disable the callback and re-call on back... So...

```kotlin
override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    activity?.onBackPressedDispatcher?.addCallback(this, object : OnBackPressedCallback(true) {
        override fun handleOnBackPressed() {
          if(isInterceptBackPress()){
            //  do your work here
          }else{
            isEnabled = false
            activity?.onBackPressed()
          }
        }
    })
}
```